src/js/components/TextInput/TextInput.js

```diff
@@ -152,18 +153,39 @@ const TextInput = forwardRef(
     }, [onSuggestionsOpen, suggestions]);
     /* eslint-enable react-hooks/exhaustive-deps */
 
+    // choose the best suggestion, either the explicit default or the one
+    // that matches the current value
+    const bestSuggestion = useCallback(() => {
+      if (suggestions) {
+        const suggestionValues = suggestions.map(suggestion =>
+          typeof suggestion === 'object' ? suggestion.value : suggestion,
+        );
+        const indexOfValue = suggestionValues.indexOf(value);
+        if (indexOfValue === -1 && typeof defaultSuggestion === 'number') {
+          return defaultSuggestion;
+        }
+        return indexOfValue;
+      }
+      return -1;
+    }, [defaultSuggestion, suggestions, value]);
+
+    // activeSuggestionIndex unifies mouse and keyboard interaction of
+    // the suggestions
     const [activeSuggestionIndex, setActiveSuggestionIndex] = useState(
-      typeof defaultSuggestion === 'number' ? defaultSuggestion : -1,
+      bestSuggestion(),
     );
 
+    // Only update active suggestion index when the mouse actually moves,
+    // not when suggestions are moving under the mouse.
+    const [mouseArmed, setMouseArmed] = useState();
```

Comment:
this variable name wasn't immediately clear to me -- maybe a different name. from my understanding, this is a way to discern that if we should be relying on the mouse placement or not to decide on the suggestion, some ideas:
- `mouseReady`
- `mouseControl`
- `mouseDetermineSuggestion`
- `mouseReadyToSelect`

no necessarily sold on some of these, but figured I'd leave as a brainstorm.

Reply:
Renamed to `mouseMovedSinceLastKey`